addons/pos_paytm/models/pos_payment_method.py

```diff
@@ -0,0 +1,97 @@
+
+    paytm_mid = fields.Char(string='PayTM Merchant ID')
+    paytm_tid = fields.Char(string='PayTM Terminal ID')
+    merchant_key = fields.Char(string='PayTM Merchant Key')
+    channel_id = fields.Char(string='PayTM Channel ID', default='EDC')
+    paytm_test_mode = fields.Boolean(string='Test Mode', help='Run transactions in the test environment.')
+    accept_payment = fields.Selection(selection=[('auto', 'Automatically'), ('manual', 'Manually')], default='auto')
+    allowed_payment_modes = fields.Selection(selection=[('all', 'All'), ('card', 'Card'), ('qr', 'QR')], default='all')
+
+    def _get_payment_terminal_selection(self):
+        selection = super()._get_payment_terminal_selection()
+        selection += [('paytm', 'PayTM')]
+        return selection
+
+    def _paytm_make_request(self, url, payload=None):
+        """ Make a request to PayTM API.
+
+        Note: self.ensure_one()
+
+        :param str url: The url to be reached by the request.
+        :param dict payload: The payload of the request.
+        :return The JSON-formatted content of the response.
+        :rtype: dict
+        """
+        self.ensure_one()
+        try:
+            response = requests.post(url, json=payload)
+            response.raise_for_status()
+        except requests.exceptions.RequestException:
+            _logger.warning("Cannot connect with PayTM")
+            return {'body':{'resultInfo': {'resultCode': "F", "resultMsg": _("Unable to establish connection with PayTM.")}}}
+        return response.json()
+
+    def paytm_make_payment_request(self, amount, transaction, timestamp):
+        body = self._paytm_get_request_body(transaction, timestamp)
+        body['transactionAmount'] = str(int(amount))
+        if self.accept_payment == 'auto':
+            body['autoAccept'] = 'True'
+        body['paymentMode'] = self.allowed_payment_modes.upper()
+        head = self._paytm_get_request_head(body)
+        merchantExtendedInfo = {'paymentMode': self.allowed_payment_modes.upper()}
+        if self.accept_payment == 'auto':
+            merchantExtendedInfo['autoAccept'] = 'True'
+        body['merchantExtendedInfo'] = merchantExtendedInfo
+        payload = {'head': head, 'body': body}
+        api_url = 'https://securegw-stage.paytm.in/ecr/payment/request' if self.paytm_test_mode else 'https://securegw-edc.paytm.in/ecr/payment/request'
+        response = self._paytm_make_request(api_url, payload=payload)
+        return response
+
+    def paytm_fetch_payment_status(self, transaction, timestamp):
+        body = self._paytm_get_request_body(transaction, timestamp)
+        head = self._paytm_get_request_head(body)
+        payload = {'head': head, 'body': body}
+        api_url = 'https://securegw-stage.paytm.in/ecr/V2/payment/status' if self.paytm_test_mode else 'https://securegw-edc.paytm.in/ecr/V2/payment/status'
+        response = self._paytm_make_request(api_url, payload=payload)
+        return response
```

Comment:
try except blalbabla
when you make a request on a server (odoo or external) think about error handling.
because here in case of problem it is guaranteed traceback